bodhi/models/models.py

```diff
@@ -1044,6 +1044,7 @@ def set_request(self, db, action, username):
         topic = u'update.request.%s' % action
         if action is UpdateRequest.unpush:
             self.unpush()
+            self.request = None
```

Comment:
Yeah, there's a *second* place that ``Update.unpush(db)`` is called.

- First there is the place where you added a the ``self.request = None`` line on 1046.
- But there is *also* a place where it is called on line 836 (when an update is edited and the builds are changed).

I *think* (and @lmacken may correct me if I'm wrong here) that ``self.request`` should be set to ``None`` in *both* circumstances.  The best way to code that I think is to remove the ``self.request = None`` line where you added it and to *instead* add it to the end of the ``Update.unpush(db)`` method on line 1550.  That way, the request will be set to ``None`` **everywhere** in the code that something tries to unpush an Update (not just when the user explicitly requests that -- which is the code path that you edited).

Does that make sense?